### PR TITLE
New Resource: wiz_connector_aws

### LINKED ISSUE #11: New Resource: wiz_connector
### Description

Add support for a new resource, wiz_connector, to manage Connectors within Wiz.

### Potential Terraform Configuration

```hcl
resource "wiz_connector" "test" {
  name        = "test"
  enabled     = true
  auth_params = jsonencode({})
  extra_config = jsonencode({
    "auditLogMonitorEnabled" : false,
    "excludedAccounts" : [],
    "excludedOUs" : []
  })
  type = jsonencode({
    "id" : "aws",
    "name" : "Amazon Web Services (AWS)",
    "categories" : [
      {
        "id" : "CloudPlatform",
        "name" : "Cloud platform"
      },
      {
        "id" : "UserDirectory",
        "name" : "User directory"
      },
      {
        "id" : "IAC",
        "name" : ""
      },
      {
        "id" : "CSPM",
        "name" : ""
      },
      {
        "id" : "SecurityInsight",
        "name" : ""
      }
    ],
    "technology" : {
      "id" : "2263",
      "name" : "AWS Account"
    },
    "authorizeUrls" : null
  })
}
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by jschoombee**
closes #11 

acceptance tests:
```sh
TF_ACC=1 go test ./internal/acceptance/... -v -run='TestAccResourceWizConnectorAws_basic'
=== RUN   TestAccResourceWizConnectorAws_basic
2023/06/28 13:48:02 [DEBUG] POST https://auth.app.wiz.io/oauth/token
2023/06/28 13:48:03 [DEBUG] POST https://auth.app.wiz.io/oauth/token
2023/06/28 13:48:03 [DEBUG] POST https://auth.app.wiz.io/oauth/token
2023/06/28 13:48:04 [DEBUG] POST https://api.eu4.app.wiz.io/graphql
2023/06/28 13:48:04 [DEBUG] POST https://api.eu4.app.wiz.io/graphql
2023/06/28 13:48:04 [DEBUG] POST https://auth.app.wiz.io/oauth/token
2023/06/28 13:48:05 [DEBUG] POST https://auth.app.wiz.io/oauth/token
2023/06/28 13:48:06 [DEBUG] POST https://api.eu4.app.wiz.io/graphql
2023/06/28 13:48:06 [DEBUG] POST https://auth.app.wiz.io/oauth/token
2023/06/28 13:48:06 [DEBUG] POST https://auth.app.wiz.io/oauth/token
2023/06/28 13:48:07 [DEBUG] POST https://api.eu4.app.wiz.io/graphql
--- PASS: TestAccResourceWizConnectorAws_basic (5.57s)
PASS
ok      wiz.io/hashicorp/terraform-provider-wiz/internal/acceptance     5.838s
```

### CODE REVIEW

File: internal/wiz/structs.go [hunk 1]
```diff
@@ -2295,4 +2345,51 @@ type ClickUpCreateTaskActionTemplateParams struct {
...
+type CreateConnectorInput struct {
+	Name        string          `json:"name"`
+	Type        string          `json:"type"` // ID Scalar
+	Enabled     *bool           `json:"enabled,omitempty"`
+	AuthParams  json.RawMessage `json:"authParams"`
+	ExtraConfig json.RawMessage `json:"extraConfig,omitempty"`
+}
+
+// CreateConnectorAuthConfigAWSInput struct
+type CreateConnectorAuthConfigAWSInput struct {
```
Comment (gramsa49) [inline internal/wiz/structs.go:2358]: I don't see this struct in the Wiz API.  If this is a custom struct, it should go into the provider resource file instead.

Comment (jschoombee) [reply internal/wiz/structs.go]: spotted, it was unreferenced and now removed

File: internal/wiz/structs.go [hunk 2]
```diff
@@ -562,10 +562,13 @@ type AzureServiceBusAutomationActionParams struct {
...
 }
 
-// Connector struct -- updates.  this resource is incomplete.  missing ConnectorConfigs ConnectorModules TenantIdentityClient
+// Connector struct -- updates.  this resource is incomplete
+// not all ConnectorConfigs are defined, missing ConnectorModules TenantIdentityClient
 type Connector struct {
-	AddedBy           User            `json:"addedBy"`
-	AuthParams        json.RawMessage `json:"authParams"`
+	AddedBy           User            `json:"addedBy"` // this has been deprecated and should be replaced by CreatedBy
+	AuthParams        interface{}     `json:"authParams"`
```
Comment (gramsa49) [inline internal/wiz/structs.go:569]: This shows as JSON in the GraphQL schema.  Can you explain why it is defined as an interface{} in the golang?

File: internal/wiz/structs.go [hunk 3]
```diff
@@ -562,10 +562,13 @@ type AzureServiceBusAutomationActionParams struct {
...
+	Config            interface{}     `json:"config,omitempty"`  // type assertion i.e. ConnectorConfigAWS
+	Outpost           interface{}     `json:"outpost,omitempty"` // type assertion i.e. OutpostAWSConfig
```
Comment (gramsa49) [inline internal/wiz/structs.go:571]: This shows as type Outpost in the GraphQL schema.  Why is it an interface{} in golang?

File: internal/provider/resource_connector_aws.go [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+	            externalID
+	            accessKey
+	            secretKey
+	            stateBucketName
+	            resultsBucketName
+	            settingsRegion
+	            disableNatGateway
+	            }
+	        }
+	    }
```
Comment (gramsa49) [inline internal/provider/resource_connector_aws.go:232]: When managing Connector resources, only the ID is required for the provider.  Is there a reason we're chasing the outpost details on read?  I'm trying to understand the value or need for this.  Let me know if I am missing something.

Comment (jschoombee) [reply internal/provider/resource_connector_aws.go:232]: The assumption was the outpost orchestrator role arn would be useful as an output for any IAM-type automations. Investigating a bit I don't see references to it in any IAM trust policies, so fair comment, this is probably not worth chasing.

File: internal/wiz/structs.go (hunk 2 above)
Comment (jschoombee) [reply internal/wiz/structs.go:569]: was with experimenting to use a single connector for all CSPs, should have been cleaned up when dropping the `CreateConnectorAuthConfigAWSInput`

**Review by gramsa49 — APPROVED** [review]

**Comment by gramsa49** [comment]
Thanks for the contribution!  Looks good to me.

**Pull request merged**